core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -301,6 +301,28 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (proxyUser != null && principal != null) {
       error("Only one of --proxy-user or --principal can be provided.")
     }
+
+    // Ensure heartbeat arguments is valid.
+    val executorTimeoutMs =
+      Utils.timeStringAsMs(sparkProperties.getOrElse(
+        "spark.storage.blockManagerSlaveTimeoutMs",
+        s"${Utils.timeStringAsSeconds(
```

Comment:
Sticking these in a string interpolation split across lines is hard to read. Just pull this into a `val`.

Reply:
good!